test/sdk/nodejs/testSDKClient.js

```diff
@@ -44,26 +46,118 @@ const connect = async () => {
 
 		await agonesSDK.reserve(5);
 
-		setTimeout(() => {
-			console.log('send allocate request');
-			agonesSDK.allocate();
-		}, 1000);
-
-		setTimeout(() => {
-			console.log('send shutdown request');
-			agonesSDK.shutdown();
-		}, 1000);
-		setTimeout( () => {
-			console.log('closing agones SDK');
-			// Closing Agones SDK and all event emitters
-			agonesSDK.close()
-		}, 2000);
-		setTimeout(() => {
-			process.exit(0);
-		}, 2000);
+		await setTimeout(1000);		
+		console.log('send allocate request');
+		agonesSDK.allocate();
+
+		await testCounts(agonesSDK);
+		await testLists(agonesSDK);
+
+		await setTimeout(1000);
+		console.log('send shutdown request');
+		agonesSDK.shutdown();
+		
+		await setTimeout(2000);
+		console.log('closing agones SDK');
+		// Closing Agones SDK and all event emitters
+		agonesSDK.close()
+		
+		await setTimeout(2000);
+		process.exit(0);
 	} catch (error) {
 		console.error(error);
 	}
 };
 
+const testCounts = async(sdk) => {
+	// LocalSDKServer starting "rooms": {Count: 1, Capacity: 10}
+	const counter = "rooms";
+
+	try {
```

Comment:
I've not used the Assert class much because it is usual to rely on a test runner framework such as jasmine or mocha instead.
In runtime code it is standard to rely on try/catch when dealing with async functions. In test code it is convenient to do the same.

I think in this case as we want to also check the return value it is easier to write the code with try/catch. We end up nesting in a try/catch block vs. nesting in a function call, which is preferable.

It may be worth investigating in another PR and also investigate if we could swap out jasmine for the new node.js built in test runner, which relies on assert.